client/state/marketplace/purchase-flow/selectors.ts

```diff
@@ -1,7 +1,16 @@
 import 'calypso/state/marketplace/init';
-import { IPurchaseFlowState } from 'calypso/state/marketplace/types';
+import {
+	IPurchaseFlowState,
+	MARKETPLACE_ASYNC_PROCESS_STATUS,
+} from 'calypso/state/marketplace/types';
 import { IAppState } from 'calypso/state/types';
 
 export function getPurchaseFlowState( state: IAppState ): IPurchaseFlowState {
-	return state.marketplace.purchaseFlow;
+	return (
+		state?.marketplace?.purchaseFlow ?? {
+			primaryDomain: null,
+			productSlugInstalled: null,
+			pluginInstallationStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
+		}
```

Comment:
`state.marketplace.purchaseFlow` would never be `undefined` and would always return the default state from the reducer, right? Maybe I am missing something here.

Reply:
Technically you are correct but we are following a convention where we safeguard against the reducer being undefined since they are/maybe registered dynamically.

https://github.com/Automattic/wp-calypso/blob/6e6636e9633456a68032d28fafd6d734e08db379/client/state/types.ts#L15-L18, so we have to guard for undefined.

Reply:
Oh okay! Maybe we can import the default state from the reducer and return that instead?
https://github.com/Automattic/wp-calypso/blob/11d41f2c0be1329084b7093e3780c2f6ad71ac76/client/state/marketplace/purchase-flow/reducer.ts#L12-L16

Reply:
Ah yes good call, for some reason this looked like a type in my head 😄